src/smc-webapp/course/actions.ts

```diff
@@ -744,26 +746,16 @@ export class CourseActions extends Actions<CourseState> {
     });
   }
 
-  delete_student(student) {
+  public async delete_student(student): Promise<void> {
     const store = this.get_store();
     if (store == null) {
       return;
     }
-    student = store.get_student(student);
-    const project_id = student.get("project_id");
-    if (project_id != null) {
-      // The student's project was created so let's clear any upgrades from it.
-      this.redux.getActions("projects").clear_project_upgrades(project_id);
-    }
-    this._set({
-      deleted: true,
-      student_id: student.get("student_id"),
-      table: "students"
-    });
+    await this._delete_student(store.get(student));
```

Comment:
Woops -- `store.get(student)` is total nonsense.  Can you make typescript catch that sort of thing?